ui/src/main/java/org/pentaho/di/ui/spoon/Spoon.java

```diff
@@ -3064,20 +3143,66 @@ public void monitorClusterSchema() throws KettleException {
     monitorClusterSchema( clusterSchema );
   }
 
+  /**
+   * Edit the Slave server configuration
+   */
   public void editSlaveServer() {
-    final SlaveServer slaveServer = (SlaveServer) selectionObject;
-    editSlaveServer( slaveServer );
+    SpoonTreeLeveledSelection leveledSelection = (SpoonTreeLeveledSelection) selectionObject;
+    editSlaveServer( leveledSelection );
+  }
+
+  private void editSlaveServer(SpoonTreeLeveledSelection leveledSelection ) {
+    SlaveServerManagementInterface slaveServerManager = null;
+    SlaveServer slaveServer = null;
+    try {
+      slaveServerManager = getSlaveServerManager( leveledSelection.getLevel() );
+      slaveServer = slaveServerManager.get( leveledSelection.getName() );
+      delegates.slaves.edit( slaveServerManager, slaveServer );
+    } catch ( Exception e ) {
+      new ErrorDialog( shell, "Error", "Unexpected error retrieving slave servers", e );
+    }
+  }
+
+  private SlaveServerManagementInterface getSlaveServerManager( LeveledTreeNode.LEVEL level ) throws KettleException {
+    if ( level == LeveledTreeNode.LEVEL.PROJECT ) {
+      return managementBowl.getManager( SlaveServerManagementInterface.class );
+    } else if ( level == LeveledTreeNode.LEVEL.GLOBAL ) {
+      return DefaultBowl.getInstance().getManager( SlaveServerManagementInterface.class );
+    } else if ( level == LeveledTreeNode.LEVEL.LOCAL ) {
+      AbstractMeta meta = getActiveAbstractMeta();
+      return meta != null ? meta.getSlaveServerManagementInterface() : null;
+    } else {
+      // shouldn't happen.
+      return null;
+    }
```

Comment:
This could almost be combined with getDBManager. Just need a way to get a generic manager from the meta.

Reply:
Yeah... but not sure how to return a generic manager from AbstractMeta. May be we need to some logic like BaseBowl.getManager() ??

Reply:
I don't think we need the registration part. Maybe just a switch/case?